providers/sum/sum.go

```diff
@@ -0,0 +1,280 @@
+		sumArgs,
+	).Info("Calling sum")
+
+	cmd := exec.CommandContext(ctx, s.SumPath, sumArgs...)
+	b_out, err := cmd.CombinedOutput()
+	if err != nil || ctx.Err() != nil {
+		if err != nil {
+			return
+		}
+
+		if ctx.Err() != nil {
+			return output, ctx.Err()
+		}
+	}
+
+	return string(b_out), err
+}
+
+func (s *Exec) GetCurrentBiosCfg(ctx context.Context) (output string, err error) {
+	return s.run(ctx, "GetCurrentBiosCfg")
+}
+
+func (s *Exec) LoadDefaultBiosCfg(ctx context.Context) (err error) {
+	_, err = s.run(ctx, "LoadDefaultBiosCfg")
+	return
+}
+
+func (s *Exec) ChangeBiosCfg(ctx context.Context, cfgFile string, reboot bool) (err error) {
+	args := []string{"--file", cfgFile}
+
+	if reboot {
+		args = append(args, "--reboot")
+	}
+
+	_, err = s.run(ctx, "ChangeBiosCfg", args...)
+
+	return
+}
+
+// GetBiosConfiguration return bios configuration
+func (s *Exec) GetBiosConfiguration(ctx context.Context) (biosConfig map[string]string, err error) {
+	biosText, err := s.GetCurrentBiosCfg(ctx)
+	if err != nil {
+		return
+	}
+
+	// We need to call vcm here to take the XML returned by SUM and convert it into a simple map
+	vcm, err := config.NewVendorConfigManager("xml", common.VendorSupermicro, map[string]string{})
+	if err != nil {
+		return
+	}
+
+	err = vcm.Unmarshal(biosText)
+	if err != nil {
+		return nil, err
+	}
+
+	biosConfig, err = vcm.StandardConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return biosConfig, nil
+}
+
+// SetBiosConfiguration set bios configuration
+func (s *Exec) SetBiosConfiguration(ctx context.Context, biosConfig map[string]string) (err error) {
+	vcm, err := config.NewVendorConfigManager("xml", common.VendorSupermicro, map[string]string{})
+	if err != nil {
+		return
+	}
+
+	for k, v := range biosConfig {
+		switch {
+		case k == "boot_mode":
+			if err = vcm.BootMode(v); err != nil {
+				return err
+			}
+		case k == "boot_order":
+			if err = vcm.BootOrder(v); err != nil {
+				return err
+			}
+		case k == "intel_sgx":
+			if err = vcm.IntelSGX(v); err != nil {
+				return err
+			}
+		case k == "secure_boot":
+			switch v {
+			case "Enabled":
+				if err = vcm.SecureBoot(true); err != nil {
+					return err
+				}
+			case "Disabled":
+				if err = vcm.SecureBoot(false); err != nil {
+					return err
+				}
+			}
+		case k == "tpm":
+			switch v {
+			case "Enabled":
+				if err = vcm.TPM(true); err != nil {
+					return err
+				}
+			case "Disabled":
+				if err = vcm.TPM(false); err != nil {
+					return err
+				}
+			}
+		case k == "smt":
+			switch v {
+			case "Enabled":
+				if err = vcm.SMT(true); err != nil {
+					return err
+				}
+			case "Disabled":
+				if err = vcm.SMT(false); err != nil {
+					return err
+				}
+			}
+		case k == "sr_iov":
+			switch v {
+			case "Enabled":
+				if err = vcm.SRIOV(true); err != nil {
+					return err
+				}
+			case "Disabled":
+				if err = vcm.SRIOV(false); err != nil {
+					return err
+				}
+			}
+		case strings.HasPrefix(k, "raw:"):
```

Comment:
The raw prefix is an ironlib thing, can we instead have method in bmc-toolbox that adds and removes this prefix, and that method is invoked here and in ironlib. This will provide more context into this prefix value.

Reply:
I'd like to handle this in a future PR.